### PR TITLE
User.authenticate

### PR DESCRIPTION
**Pull request by zhuchini**
I've implemented all the necessary features, however logging in is causing an issue.
I've tried creating a new 'authenticate' method because the error shows "undefined method 'authenticate' for nil:NilClass," but that didn't work.

### CODE REVIEW

**Comment by zhuchini** [comment]
Rubocop is telling me to use 'if' with a negated condition when I use 'unless,' and vice versa.

File: app/models/event.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  has_many :attendances, foreign_key: 'attended_event_id'
+  has_many :attendees, through: :attendances
+
+  validates :title, presence: true
+  validates :location, presence: true
+  validates :date, presence: true
+
+  default_scope { order(date: :desc) }
+  scope :previous_events, -> { where('date < ?', Time.now) }
+  scope :upcoming_events, -> { where('date > ?', Time.now) }
```
Comment (mcrd25) [inline app/models/event.rb]: I see that you've implemented your scopes but you have not used them in your controllers for getting past and upcoming events which is a requirement as can be seen from this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/past_upcoming_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations) and this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from same

**Comment by zhuchini** [comment]
I apologize in advance, but moving all the methods in the SessionsHelper to my ApplicationController does not help my project's functionality because I still need to add a 'helper_method' so that the methods in my ApplicationController have access to all files.

I'm simply following the examples and tutorials given to us by The Odin Project.

sessionshelper

sessionshelper2

**Comment by zhuchini** [comment]
Also, I've spoken with Willow that I can work alone on the Ruby on Rails section because my partner left the program.

**Comment by mcrd25** [comment]
Hi, @zhuchini  I think you misunderstood the helper comments.

Methods that specific to views should remain in helper. those that are meant for application logic should be in application controller. Does this make sense? 😅 
Also from your  PR description are you saying this issue still persists?

**Comment by zhuchini** [comment]
I apologize for my stubbornness, I must have not eaten that day.
I moved all the methods in the Sessions_Helper to my Application_controller file.

**Pull request merged**